Gemfile

```
@@ -96,3 +101,8 @@ end
 #
 # TODO: Remove this once the issue is resolved: https://github.com/ffi/ffi/issues/1107
 gem 'ffi', '~> 1.16.3', require: false
+
+# Ruby 3.4 should be supported by strscan v3.1.1. However, the latest release of strscan is v3.1.0.
+#
+# TODO: Remove this v3.1.0 requirement once v3.1.1 is released.
+gem 'strscan', '3.1.0' if RUBY_VERSION.start_with?('3.4.')
```

Ruby 3.4 automatically uses v3.1.1 of `gem 'strscan'` ([see here](https://stdgems.org/strscan/)), but v3.1.0 is the latest release of `strscan` ([see here](https://rubygems.org/gems/strscan)). After the release of v3.1.1, I assume this version requirement can be removed.

---

Maybe instead of forcing the usage of 3.1.0, we could use the git repository, so if Ruby 3.4 is using any new feature or bugfix that has been added since the release of 3.1.0, it should still work

 ```suggestion
gem 'strscan', git: 'https://github.com/ruby/strscan' if RUBY_VERSION.start_with?('3.4.')
```

---

Good idea! Will do this